src/tests/pools/D3MCompoundPool.t.sol

```diff
@@ -0,0 +1,282 @@
+        z = _mul(x, WAD) / y;
+    }
+
+    function _assertEqApprox(uint256 _a, uint256 _b) internal {
+        uint256 a = _a;
+        uint256 b = _b;
+        if (a < b) {
+            uint256 tmp = a;
+            a = b;
+            b = tmp;
+        }
+        if (a - b > b / 10 ** 9) {
+            emit log_bytes32("Error: Wrong `uint' value");
+            emit log_named_uint("  Expected", _b);
+            emit log_named_uint("    Actual", _a);
+            fail();
+        }
+    }
+
+    function setUp() public override {
+        hevm = Hevm(
+            address(bytes20(uint160(uint256(keccak256("hevm cheat code")))))
+        );
+
+        contractName = "D3MCompoundPool";
+
+        dai         = DaiLike(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+        cDai        = CErc20Like(0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643);
+        comptroller = ComptrollerLike(0x3d9819210A31b4961b30EF54bE2aeD79B9c9Cd3B);
+        comp        = TokenLike(0xc00e94Cb662C3520282E6f5717214004A7f26888);
+        lens        = LensLike(0xdCbDb7306c6Ff46f77B349188dC18cEd9DF30299);
+
+        vat = address(new FakeVat());
+        hub = address(new FakeHub(vat));
+
+        d3mTestPool = address(new D3MCompoundPool(hub, address(cDai)));
+        pool = D3MCompoundPool(d3mTestPool);
+
+        // allocate some dai for the pool
+        _giveTokens(dai, 100 * WAD);
+        dai.transfer(address(pool), 100 * WAD);
+    }
+
+    function test_sets_dai_value() public {
+        assertEq(address(pool.dai()), address(dai));
+    }
+
+    function test_sets_cdai_value() public {
+        assertEq(address(pool.cDai()), address(cDai));
+    }
+
+    function test_sets_comptroller_value() public {
+        assertEq(address(pool.comptroller()), address(comptroller));
+    }
+
+    function test_can_file_king() public {
+        assertEq(address(pool.king()), address(0));
+
+        pool.file("king", address(123));
+        assertEq(address(pool.king()), address(123));
+    }
+
+    function test_cannot_file_king_no_auth() public {
+        pool.deny(address(this));
+        assertRevert(d3mTestPool, abi.encodeWithSignature("file(bytes32,address)", bytes32("king"), address(123)), "D3MCompoundPool/not-authorized");
+    }
+
+    function test_cannot_file_king_vat_caged() public {
+        FakeVat(vat).cage();
+        assertRevert(d3mTestPool, abi.encodeWithSignature("file(bytes32,address)", bytes32("king"), address(123)), "D3MCompoundPool/no-file-during-shutdown");
+    }
+
+    function test_deposit_calls_cdai_deposit() public {
+        assertEq(cDai.balanceOfUnderlying(address(pool)), 0);
+        assertEq(cDai.balanceOf(address(pool)), 0);
+        uint256 poolBefore = dai.balanceOf(address(pool));
+
+        pool.file("hub", address(this));
+        pool.deposit(1 * WAD);
+
+        _assertEqApprox(cDai.balanceOfUnderlying(address(pool)), 1 * WAD);
+        assertGt(cDai.balanceOf(address(pool)), 0);
```

Comment:
This is pointless checking the one below.

Reply:
It does make sure that `_wdiv(1 * WAD, cDai.exchangeRateStored())` does not result in 0 and would go unnotice in this test.
Are you convinced it's worthless?

Reply:
Yeah I thought that might be the reason why was added, however I do not see a practical way than `_wdiv(1 * WAD, cDai.exchangeRateStored()` can be absolute 0.
Anyway this was a comment that honestly I should not have even done hehe.

Reply:
Let's leave it as it is.